### ISSUE TITLE
Node/Pool Work and Crash Dump

### ISSUE THREAD

**Issue by Mandelhoff** [zack-bitcoin/amoveo#94]
I got my node synced and then started my local pool. It appears to have worked, but gave some odd console output:

![image](https://user-images.githubusercontent.com/35471470/35835477-9849512e-0aa0-11e8-87b0-2b7423092684.png)

I mined two blocks and got this crash dump on the pool and a bunch of "weir responses" on the node:

![image](https://user-images.githubusercontent.com/35471470/35835502-b8fea81a-0aa0-11e8-9b9e-9240c6e55255.png)

I think the "mine" might have worked though. My node wallet got some veo:
BIJw6vdb+BexSLCZQ7XZEsW6RDltxTrmbGkDbKb1yNPhOpngyKDAqBkGMKaa1ZEH9Bh2d7ezhTX6nhGBldFzEPU=

Are these errors or just scary console outputs?

**Comment by zack-bitcoin** [comment]
The "status report" on the mining pool is unexpected. I had never seen it until just now, but I successfully reproduced it.
The amoveo full node is getting the normal messages from the mining pool, so I think it is still working normally.
The error messages on your full node, I just did a pull request so that the full node can handle having more than one peer without crashing so much. Probably this will solve at least part of your problem.

I think the node is still functioning correctly and maintaining consensus, even though it is having some communication trouble.
